### ISSUE TITLE
`Serial` field won't work in SQLite if it is not the primary key

### ISSUE THREAD

**Issue by metakot** [piccolo-orm/piccolo#981]
piccolo==1.5.0
piccolo_admin==1.3.2
piccolo_api==1.4.0

So I have a table in SQLite:
```python
class Example(Table):
    number = Serial()
```
When trying to save it
```python
example = Example()
await example.save()
```
the sqlite error occuring:
```
sqlite3.IntegrityError: NOT NULL constraint failed: example.number
```
Apparently piccolo creates `serial` field as just `INTEGER` without `AUTOINCREMENT` which is needed by SQLite.

**Comment by sinisaos** [comment]
I don't think this is a bug, but rather normal SQLite behavior. As stated in the [SQLite documentation](https://www.sqlite.org/autoinc.html), `AUTOINCREMENT is not allowed on WITHOUT ROWID tables or on any column of the table other than an INTEGER PRIMARY KEY. Any attempt to use AUTOINCREMENT on a WITHOUT ROWID table or on a column other than an INTEGER PRIMARY KEY column results in an error`. 
The correct way to use Serial column in SQLite is only for a primary key like this
```python
class Example(Table):
    number = Serial(primary_key=True)
```
or if it is not a primary key.
```python
class Example(Table):
    number = Integer()
```
If I'm wrong feel free to reopen issue.
